draft-ietf-regext-rdap-geofeed.md

```diff
@@ -32,107 +32,235 @@ email = "tomh@apnic.net"
-  "endAddress" : "2001:db8:0:ffff:ffff:ffff:ffff:ffff",
-  "ipVersion" : "v6",
-  "name": "NET-RTR-1",
-  "type" : "DIRECT ALLOCATION",
-  "country" : "AU",
-  "parentHandle" : "YYYY-RIR",
-  "status" : [ "active" ],
-  …
-  "geofeedv1_geofeed" : "https:example.net/geofeed"
+    "objectClassName": "ip network",
+    "handle": "XXXX-RIR",
+    "startAddress": "2001:db8::",
+    "endAddress": "2001:db8:0:ffff:ffff:ffff:ffff:ffff",
+    "ipVersion": "v6",
+    "name": "NET-RTR-1",
+    "type": "DIRECT ALLOCATION",
+    "country": "AU",
+    "parentHandle": "YYYY-RIR",
+    "status": [ "active" ],
+    "links":
+     [
+        {
+            "value": "https://example.net/ip/2001:db8::/48",
+            "rel": "self",
+            "href": "https://example.net/ip/2001:db8::/48",
+            "type": "application/rdap+json"
+        },
+        ...
+    ],
+    "redacted":
+    [
+        {
+            "name":
+            {
+                "description": "Geofeed links"
+            },
+            "prePath": "$.links[?(@.rel=='geo')]",
+            "method": "removal"
+        }
+    ],
+    ...
 }
 ```
 
 # Privacy Considerations
 
-When including a geofeed file URL in an IP Network object, an RDAP server
-operator SHOULD follow the guidance from [@?I-D.ymbk-opsawg-9092-update, section 7]
-to not accidentally expose the location of an individual. 
+When including a geofeed file URL in an IP Network object, an RDAP server operator SHOULD follow the guidance from
+[@?I-D.ymbk-opsawg-9092-update, section 7] to not accidentally expose the location of an individual. 
 
 # Security Considerations
 
-[@?I-D.ymbk-opsawg-9092-update] requires an HTTPS URL for a geofeed file, and
-optionally RPKI-signing the data within. Besides that, this document does not
-introduce any new security considerations past those already discussed in the
+[@?I-D.ymbk-opsawg-9092-update] requires an HTTPS URL for a geofeed file, and optionally RPKI-signing the data within.
+Besides that, this document does not introduce any new security considerations past those already discussed in the
 RDAP protocol specifications.
 
 # IANA Considerations
 
-IANA is requested to register the following value in the RDAP Extensions
-Registry:
+## RDAP Extensions Registry
+
+IANA is requested to register the following value in the RDAP Extensions Registry:
+
+* Extension identifier: geofeed1
+* Registry operator: Any
+* Published specification: This document.
+* Contact: IETF <iesg@ietf.org>
+* Intended usage: This extension describes version 1 of a method to access the IP geolocation feed data through RDAP.
+
+## Link Relations Registry
+
+IANA is requested to register the following value in the Link Relations Registry:
+
+* Relation Name: geo
+* Description: Indicates that the link context has a resource with geographic information at the link target.
+* Reference: This document.
+
+## Media Types Registry
+
+* Type name: application
+* Subtype name: geofeed+csv
```

Comment:
Just in case, need to ask this. :). Should it be ok to register "application/geofeed+csv" when only "text/csv" exists and not "application/csv"? Would it be better to register "text/geofeed+csv" instead? Per https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types, looks like we should be ok with "application/geofeed+csv".

Reply:
I think "application" is preferable to "text" as the top-level type, because even though this can be read by an end user as-is, it will generally make more sense for the application to do some sort of processing on it first (e.g. by having functionality to support finding the corresponding geofeed entry for a given IP address), and in this sense it's analogous to "application/rdap+json".

As far as "application/csv" goes, I can't see any text that indicates that this would be necessary, and its absence is unlikely to cause confusion for anybody, so I'd just leave it out for now.

Reply:
Ok. At least we are aware.